src/main.rs

```diff
@@ -39,26 +44,23 @@ fn query_watchman(epoch_nanoseconds: u64) -> Fallible<()> {
         .context("Couldn't start watchman")?;
 
     {
-        let time_seconds = epoch_nanoseconds / 1_000_000_000;
+        // the token following `since` expression can be either epoch second as integer or a clock id as string
+        let epoch_nanoseconds: u64 = token.parse().unwrap_or(0);
+        let mut token_value = Value::from(token);
+        if epoch_nanoseconds != 0 {
+            token_value = Value::from(epoch_nanoseconds / 1_000_000_000);
+        }
 
         let watchman_query = json!(
             [
                 "query",
                 git_work_tree,
                 {
-                    "since": time_seconds,
+                    "since": token_value,
                     "fields": ["name"],
                     "expression": [
                         "not", [
-                            "allof",[
-                                "since",
-                                time_seconds,
-                                "cclock"
-                            ],
-                            [
-                                "not",
-                                "exists"
-                            ]
+                            "dirname", ".git"
```

Comment:
Is this strictly backwards compatible with v1? I'd like to keep the existing behavior for those still on v1. Might it be a good idea to make a `query_watchman_v1` and `query_watchman_v2`?

Reply:
I had the same thought, but I tested both without problems so I used the same query.
But yeah, probably it's better to keep them separate